src/Workspaces/Core/Portable/FindSymbols/FindReferences/BaseTypeFinder.cs

```diff
@@ -2,71 +2,61 @@
-        }
+        public static ImmutableArray<ISymbol> FindBaseTypesAndInterfaces(INamedTypeSymbol type)
+            => FindBaseTypes(type).AddRange(type.AllInterfaces).CastArray<ISymbol>();
 
-        public static async Task<ImmutableArray<SymbolAndProjectId>> FindOverriddenAndImplementedMembersAsync(
+        public static ImmutableArray<ISymbol> FindOverriddenAndImplementedMembers(
             ISymbol symbol, Project project, CancellationToken cancellationToken)
         {
             var solution = project.Solution;
-            var results = ArrayBuilder<SymbolAndProjectId>.GetInstance();
-            var interfaceImplementations = ArrayBuilder<ISymbol>.GetInstance();
+            var results = ArrayBuilder<ISymbol>.GetInstance();
 
             // This is called for all: class, struct or interface member.
-            interfaceImplementations.AddRange(symbol.ExplicitOrImplicitInterfaceImplementations());
+            results.AddRange(symbol.ExplicitOrImplicitInterfaceImplementations());
 
             // The type scenario. Iterate over all base classes to find overridden and hidden (new/Shadows) methods.
             foreach (var type in FindBaseTypes(symbol.ContainingType))
             {
                 foreach (var member in type.GetMembers(symbol.Name))
                 {
-                    var sourceMember = await SymbolFinder.FindSourceDefinitionAsync(
-                        SymbolAndProjectId.Create(member, project.Id),
-                        solution,
-                        cancellationToken).ConfigureAwait(false);
+                    cancellationToken.ThrowIfCancellationRequested();
 
-                    if (sourceMember.Symbol != null)
+                    // Add to results overridden members only. Do not add hidden members.
+                    if (SymbolFinder.IsOverride(solution, symbol, member, cancellationToken))
                     {
-                        // Add to results overridden members only. Do not add hidden members.
-                        if (SymbolFinder.IsOverride(solution, symbol, sourceMember.Symbol, cancellationToken))
-                        {
-                            results.Add(sourceMember);
-                        }
+                        results.Add(member);
 
-                        // For both overridden and inherited members, 
-                        // find all explicit and implicit interface implementations.
-                        // We need to start from each base class for cases like N() Implements I.M() 
-                        // where N() can be hidden or overwritten in a nested class later on.
-                        interfaceImplementations.AddRange(member.ExplicitOrImplicitInterfaceImplementations());
+                        // We should add implementations only for overridden members but not for hidden ones.
+                        // In the following example:
+                        // interface I { void M(); }
```

Comment:
Can we add blank comment lines before/after the code snippets so it's easier to read?